extensions/script-libraries/semver/semver.livecodescript

```diff
@@ -649,38 +649,298 @@ private function __SatisfiesRange pVersion, pRange
+      switch pRangeComparator["operator"]
+         case ">"
+            if tMinorIsX then
+               -- >1 --> >=2.0.0
+               put pRangeComparator["major"] + 1 into tExpandedRange[1]["major"]
+               put 0 into tExpandedRange[1]["minor"]
+               put 0 into tExpandedRange[1]["patch"]
+               put ">=" into tExpandedRange[1]["operator"]
+            else
+               -- >1.2 --> >=1.3.0
+               put pRangeComparator["major"] into tExpandedRange[1]["major"]
+               put pRangeComparator["minor"] + 1 into tExpandedRange[1]["minor"]
+               put 0 into tExpandedRange[1]["patch"]
+               put ">=" into tExpandedRange[1]["operator"]
+            end if
+            break
+            
+         case "<="
+            if tMinorIsX then
+               -- <=1 --> <2.0.0
+               put pRangeComparator["major"] + 1 into tExpandedRange[1]["major"]
+               put 0 into tExpandedRange[1]["minor"]
+               put 0 into tExpandedRange[1]["patch"]
+               put "<" into tExpandedRange[1]["operator"]
+            else
+               -- <=0.5 --> <0.6.0
+               put pRangeComparator["major"] into tExpandedRange[1]["major"]
+               put pRangeComparator["minor"] + 1 into tExpandedRange[1]["minor"]
+               put 0 into tExpandedRange[1]["patch"]
+               put "<" into tExpandedRange[1]["operator"]
+            end if
+            break
+            
+         default
+            put pRangeComparator into tExpandedRange[1]
+            if tMinorIsX then
+               put 0 into tExpandedRange[1]["minor"]
+            end if
+            if tPatchIsX then
+               put 0 into tExpandedRange[1]["patch"]
+            end if 
+            break
+      end switch
+   else if tMinorIsX then
+      -- 1 --> >=1.0.0 <2.0.0
+      put pRangeComparator["major"] into tExpandedRange[1]["major"]
+      put 0 into tExpandedRange[1]["minor"]
+      put 0 into tExpandedRange[1]["patch"]
+      put ">=" into tExpandedRange[1]["operator"]
+      
+      put pRangeComparator["major"] + 1 into tExpandedRange[2]["major"]
+      put 0 into tExpandedRange[2]["minor"]
+      put 0 into tExpandedRange[2]["patch"]
+      put "<" into tExpandedRange[2]["operator"]
+   else -- tPatchIsX
+      -- 1.2 --> >=1.2.0 <1.3.0
+      put pRangeComparator["major"] into tExpandedRange[1]["major"]
+      put pRangeComparator["minor"] into tExpandedRange[1]["minor"]
+      put 0 into tExpandedRange[1]["patch"]
+      put ">=" into tExpandedRange[1]["operator"]
+      
+      put pRangeComparator["major"] into tExpandedRange[2]["major"]
+      put pRangeComparator["minor"] + 1 into tExpandedRange[2]["minor"]
+      put 0 into tExpandedRange[2]["patch"]
+      put "<" into tExpandedRange[2]["operator"]
+   end if
+   
+   return tExpandedRange
+end __ExpandXRanges
 
-command semverParseRange @xRange
-   __SemverSplitRange xRange
+command semverParseRange @xRange, pExpand
```

Comment:
@livecodemichael this command should be documented if it's public

Reply:
Aha! looks like that's my fault originally! Sorry do you mind fixing my error here and adding the docs?